tests/test_cli_containerbuild.py

```diff
@@ -0,0 +1,558 @@
+                 task_result={},
+                 priority=None,
+                 channel=cli_containerbuild.DEFAULT_CHANNEL,
+                 running_in_background=False):
+    """
+    Mock a session for the purposes of cli_containerbuild.handle_build()
+
+    The default argument values are set up for a successful build,
+    as long as `target` and `source` match those provided via CLI args.
+    """
+    (flexmock(cli_containerbuild)
+        .should_receive('activate_session'))  # and do nothing
+    (flexmock(cli_containerbuild)
+        .should_receive('_running_in_bg')
+        .and_return(running_in_background))
+    (flexmock(cli_containerbuild)
+        .should_receive('watch_tasks')
+        .and_return(0 if task_success else 1))
+
+    build_target = {'dest_tag': dest_tag_name,
+                    'dest_tag_name': dest_tag_name}
+
+    dest_tag = {'name': dest_tag_name,
+                'locked': dest_tag_locked}
+
+    session_status = {'logged_in': True}
+
+    def logout():
+        session_status['logged_in'] = False
+
+    session = flexmock(status=session_status, logout=logout)
+    (session
+        .should_receive('getBuildTarget')
+        .with_args(target)
+        .and_return(build_target if target_known else None))
+    (session
+        .should_receive('getTag')
+        .with_args(build_target['dest_tag'])
+        .and_return(dest_tag if dest_tag_known else None))
+    (session
+        .should_receive('buildContainer')
+        .with_args(source, target, dict,
+                   priority=priority, channel=channel)
+        .and_return(task_id))
+    (session
+        .should_receive('getTaskResult')
+        .with_args(task_id)
+        .and_return(task_result))
+
+    return session
+
+
+def build_cli_args(target,
+                   source,
+                   git_branch='random-branch',
+                   wait=None,
+                   background=False):
+    """Build command line arguments for cli_containerbuild.handle_build()"""
+    args = [target, source, '--git-branch', git_branch]
+    if wait is not None:
+        args.append('--wait' if wait else '--nowait')
+    if background:
+        args.append('--background')
+    return args
+
+
+def _expected_output(result, offset, indent):
+    if isinstance(result, list):
+        for item in result:
+            for line in _expected_output(item, offset+indent, indent):
+                yield line
+    elif isinstance(result, dict):
+        for key, value in result.items():
+            yield '{}{}:\n'.format(offset, key)
+            for line in _expected_output(value, offset+indent, indent):
+                yield line
+    else:
+        yield '{}{}\n'.format(offset, result)
+
+
+def expected_task_output(task_id, result, weburl, quiet=False):
+    try:
+        result['koji_builds'] = [
+            '{}/buildinfo?buildID={}'.format(weburl, build_id)
+            for build_id in result.get('koji_builds', [])
+        ]
+    except TypeError:
```

Comment:
That's a good question. `cli_containerbuild.print_task_result()` does the same thing, I thought it was for cases when the result is not a dict, but then it would fail with `AttributeError` anyway. I'm only testing cases where the result is a dict, not sure if that is guaranteed.

Reply:
Drop that exception handling please